programs/address-lookup-table/src/processor.rs

```diff
@@ -311,7 +311,7 @@ impl Processor {
             for new_address in new_addresses {
```

Comment:
While we are at it: How about `table_data.reserve()` before the `for` loop extending it multiple times?

Reply:
I had this in a separate PR as jeff had asked me to split it, but now put it back here. I added `BorrowAccount::extend_from_slice` so the old data isn't thrown away anymore and it's just extended.